### PR TITLE
use 'execvp' instead of 'system' (fixes security vulnerabilities with 'system')

### PR DESCRIPTION
**Pull request by Itai-Nelken**
The system function has a security vulnerability that makes it possible to make it run malicious code (With aliases and custom functions I think).
I changed to 'execvp()'. there is no need to detect user's shell as execvp() will search for the command in PATH (or if you supply a path, it will use it).

The comments I wrote in the code explain it, but here is a small diagram of what happens:
```
____________________________________
|                                  |  <-- child PID
| DollarSKip parent (main) process | -- fork() --|
-----------------------------------              |
                  |                              \/
                  |                   exact copy of parent process
                  |                       ---------------
                  |                      |               | <-- child_pid is 0 here because 
                  |                      | child process |     child has no child process
                  |                       ---------------
                  |                             |
                  |                             |
               execvp()------------------------>|the child process gets replaced with the command
                  |                             | execvp inserts
          parent process waits                  |
       until the child has finished        child finished
            with waitpid()                 =============
                  |
              parent ends
              ==========
```
hope this helps to understand what the code does.

Signed-off-by: Itai Nelken <itainelken@gmail.com>

### CODE REVIEW

**Comment by Itai-Nelken** [comment]
I'm not sure the diagram makes it clear, but as `child_pid` is 0 in the child process, `if(child_pid==0)` will be true, so execvp() will be called.
if it's smaller than 0, fork() has failed, else its the parent process, so wait until the child exits.

**Comment by CleanMachine1** [comment]
Does this still need to be reviewed or close?

**Comment by Itai-Nelken** [comment]
Review the code if you want.
Even if this pr doesn't fix #19, it resolves the security vulnerability of running commands through a shell (what system()) by making the operaying system run the executable directly.

**Comment by Itai-Nelken** [comment]
It needs testong though as it might break shell commands

**Comment by CleanMachine1** [comment]
Alright. 

I think we should make a new make test type of command, to automate it just running loads of commands

**Comment by CleanMachine1** [comment]
Also I would just like to say, this would be v3 of DollarSkip and v2 is still in "beta"

**Comment by Itai-Nelken** [comment]
I think this should go into v2, my other idea (use `/path/to/shell -c 'command'`, the current v2) is pretty bad TBH.

also a problem that exists in all versions but only this pr makes it shows an error: you can't set shell variables and functions because in v1 and current v2 the shell used is separate from the shell you are using in the terminal, and in this pr it gives an error as the system is called to execute the program, it never goes through a shell.

**Comment by CleanMachine1** [comment]
I see how it could be a bad idea. I really wish we had more contributors, not for editing, but for testing

**Comment by CleanMachine1** [comment]
I'll get round to all of this slowly, since I still need to test, which is a little bit boring and long for me to do.

I really appreciate all these PRs, you are helping more than me!

**Comment by Itai-Nelken** [comment]
>I really appreciate all these PRs, you are helping more than me!
You are welcome :)

File: dollarskip.c [hunk 1]
```diff
@@ -1,25 +1,63 @@
...
+    char *separator=" ";
+    //will hold the parsed output from strtok
+    char *parsed;
+    int index=0;
+
+    //strtok returns the first word in 'input' before the space
+    parsed=strtok(input, separator);
+    //while that word isn't NULL
+    while(parsed!=NULL) {
+        command[index]=parsed; //copy it to command[index]
```
Comment (Itai-Nelken) [inline dollarskip.c:21]: We could use `realloc()` to grow the `command` variable if the input exceeds 8096 characters, that way the only thing that will stop crazy long commands is the amount of free memory available.
What do you say @CleanMachine1?

something like this:
(`max_size` is a `int` that starts with the value of 8096)
this is between line 20 and 21.
```c
if(index==max_size) {
        max_size += 200;
        command=realloc(command, sizeof(char *)*max_size);
}
```

Comment (CleanMachine1) [reply dollarskip.c:21]: Oh, sorry I think I missed this. I mean I don't think many actually ever type a 8096 character command however it it doesn't impact the usage of the command *at all* then, Yes definitely.

Comment (Itai-Nelken) [reply dollarskip.c:21]: it will make it slower, however it really won't be noticeable.

**Comment by Itai-Nelken** [comment]
Why did you close?

**Pull request closed without merging**